src/modules/security/index.ts

```diff
@@ -0,0 +1,89 @@
+import type { Faker } from '../..';
+import { toDate } from '../../internal/toDate';
+
+export interface Cvss {
+  score: number;
+  vector: string;
+  rating: 'none' | 'low' | 'medium' | 'high' | 'critical';
+}
+
+export class Security {
+  constructor(private readonly faker: Faker) {
+    // Bind `this` so namespaced is working correctly
+    for (const name of Object.getOwnPropertyNames(Security.prototype)) {
+      if (name === 'constructor' || typeof this[name] !== 'function') {
+        continue;
+      }
+      this[name] = this[name].bind(this);
+    }
+  }
+
+  /**
+   * Generates a random CVE between the given boundaries
+   *
+   * @param options
+   * @param options.from The early date boundary
+   * @param options.to The late date boundary
+   *
+   * @example
+   * faker.security.cve() // 'CVE-2011-0762'
+   * faker.security.cve({from:'2020-01-01T00:00:00.000Z', to: '2030-01-01T00:00:00.000Z') // 'CVE-2028-0762'
+   */
+  cve(options?: {
+    from: string | Date | number;
+    to: string | Date | number;
+  }): string {
+    const fromMs = toDate(options?.from || '1999-01-01T00:00:00.000Z');
+    const toMs = toDate(options?.to);
+
+    return [
+      'CVE',
+      // Year
+      this.faker.date.between(fromMs, toMs).getFullYear(),
+      // Sequence in the year
+      this.faker.random.numeric(5, { allowLeadingZeros: true }),
+    ].join('-');
+  }
+
+  /**
+   * Generates a random CWE
```

Comment:
```suggestion
   * Generates a random CWE.
```

Reply:
Maybe also add the long name for the abbreviation.

```suggestion
   * Generates a random CWE (Common Weakness Enumeration) number.
```

Is it a number?

Reply:
You're right here @ST-DDT, this function generates something which matches the format for CWE but doesn't perform any validation on whether that value is a recognised identifier. I would welcome some guidance here on the degree of accuracy to pursue, for instance:
1. There are only 1388 recognised ids at the moment should a max limit be applied here so that this method can not generated an idea > CWE-1388? The implication of applying this limit is that we would need to track new IDs being added to the external CWE register. 
2. Some identifiers have been deprecated, so maybe there should be filtered out as possible responses.

Reply:
My comment was more focussed on the jsdoc comment, but if there is a max value then we can use that. There is no need to support all CWE numbers. As for filtering deprecated ones. I wouldnt go that far.

Reply:
Original comments on this thread around JSDoc have been addressed, thanks for the time both.